#! /bin/sh /usr/share/dpatch/dpatch-run
## 66_enlarge-dh-parameters-size.dpatch by Marc Haber <mh+debian-packages@zugschlus.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Enlarge default server side size of DH parameters to 2048 from 1024.
## DP: This patch has no effect if building against gnutls >= 2.12, because
## DP: exim is using gnutls_sec_param_to_pk_bits() to get correct number
## DP: of dh_bits when built against newer gnutls-versions.

diff -NurBbp exim-4.80.orig/src/tls-gnu.c exim-4.80/src/tls-gnu.c
--- exim-4.80.orig/src/tls-gnu.c	2012-05-19 01:17:38.000000000 +0200
+++ exim-4.80/src/tls-gnu.c	2012-05-20 12:01:24.000000000 +0200
@@ -159,7 +159,7 @@ callbacks. */
 can ask for a bit-strength.  Without that, we stick to the constant we had
 before, for now. */
 #ifndef EXIM_SERVER_DH_BITS_PRE2_12
-#define EXIM_SERVER_DH_BITS_PRE2_12 1024
+#define EXIM_SERVER_DH_BITS_PRE2_12 2048
 #endif
 
 #define exim_gnutls_err_check(Label) do { \
